### ISSUE TITLE
multiQC error on empty files?

### ISSUE THREAD

**Issue by bgruening** [galaxyproject/tools-iuc#3313]
Module 'samtools: 'This file was produced by samtools stats' not found in the file 'ERR4707731' to fix in multiqc. Especially with a large collection with 4000 files multiqc fails as soon as "one" (?) file is empty?

**Comment by mblue9** [comment]
I had a look at this & it still exists in v1.9 because the wrapper has this https://github.com/galaxyproject/tools-iuc/blob/a6ce777a070951162b93312bb8e6d14324cd82df/tools/multiqc/multiqc.xml#L15

**Comment by wm75** [comment]
Yes, I've fixed to other such cases in #3241 but each tool has its own output patterns, which makes it hard to catch everything in one PR ::(
I remember vaguely that somebody reported sth similar for HISAT2 not that long ago, but I forgot where and I don't have "empty" HISAT2 output to test with. Would just be nice to fix at least two things at once.

**Comment by mblue9** [comment]
Do you think we could let MultiQC handle the pattern finding and remove the checks from the wrapper?  As I just tested removing that line above and with a collection containing one empty output it no longer fails, it just doesn't give output for the empty file.

**Comment by wm75** [comment]
Ok, I've found the HISAT2 report in the .eu bug report list so we should be able to grab an example from an existing history.
@mblue9 do you have sample output for samtools stats run on empty input? With that we could adjust the grep pattern.

**Comment by wm75** [comment]
Sorry, overlapping posting. It would be best if we could just find a grep pattern that works for "empty" *and* regular output. This way it's easier to debug user issues when they maybe just connect wrong input to multiqc.

**Comment by mvdbeek** [comment]
It's probably also worth opening an issue upstream in multiqc to see if that's something that can be fixed more generally. This doesn't seem like it's only an issue for the Galaxy wrapper ?

**Comment by wm75** [comment]
I know only very little about using multiqc on the command line, but what I understand is that it scans a directory for tool outputs it recognizes, then produces a report based on these.
So, in principle, the wrapper could just let multiqc do its job, and if a user, unintentionally, feeds the wrong input to the tool, they would get a report for that. However, the wrapper also customizes the reports based on the user's claim of what the input is and if the input was passed through to multiqc unchecked, then these customizations and the actual report wouldn't fit together.
So I believe this isn't really anything that typically affects multiqc command line users, but we need to get the Galaxy-specific checks right (unless we would want to do a major rewrite of the wrapper).

**Comment by mblue9** [comment]
@wm75 I don't have real empty samtools stats just one I faked by deleting the output in collection here https://usegalaxy.eu/u/mblue9/h/multiqc-samtools-empty but maybe @bgruening does
Could/should the wrapper give a warning instead of erroring if the pattern is not found or file is empty?

**Comment by wm75** [comment]
@bgruening This is strange. I just let samtools stats run on an empty BAM file and it *does* produce the pattern that MultiQC searches for. MultiQC also generates a report though *inside* of it a plot error gets reported.

I hope you can find the offending samtools stats output in your giant collection?
